src/app/main/main.component.ts

```diff
@@ -162,6 +169,18 @@ export class MainComponent implements OnInit, OnDestroy {
     this.updateStyles();
   }
 
+  saveDataToLocalStorage(projectID : string, sql : string, location : string) {
+    this._storage.set(this.projectLocalStorageKey, projectID);
+    this._storage.set(this.sqlLocalStorageKey, sql);
+    this._storage.set(this.locationLocalStorageKey, location);
+  }
```

Comment:
A bit of an edge case, but let's store a user ID as well? If I log out and then log back in with another account, I'll get the previous user's localStorage data here. We should also clear localStorage when the user logs out, although that's not 100% because we won't know when the session expires.

Could also consider making this whole thing a single entry in localstorage, like:

```js
this._storage.set(LOCAL_STORAGE_KEY, JSON.stringify({userID, projectID, sql, location}));
```

Reply:
Done. I had to do a bit more:

1- I'm deleting the storage during signin as well. While not 100% fool proof, it makes the session case expiration a bit better since the user has to signin again.

2- I had to reset other UI components for a case where user signs out and immediately signs back in without relaunching the app.

Reply:
I can't really think of a case where (1) isn't enough, so maybe storing user ID isn't necessary? But your comment suggests you did both and I only see one here, so just wanted to confirm that's intentional.

Reply:
- During singin and signout, I delete the local storage.

- I reset the UI fields (the state), only at signin.

Reply:
My suggestion earlier was to store userID. I'm not sure if you're saying these changes have the same effect?

Reply:
I missed the UserID part. These changes have the same effect i think. Do you think we need the userID for any other purpose?

Reply:
Looks OK to me. 👍